src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useMemo, useCallback } from 'react';
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
+import { Person } from '../types/Person';
+
+type Props = {
+  people: Person[];
+  delay?: number;
+  onSelect?: (person: Person | null) => void;
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  delay = 1000,
+  onSelect = () => {},
+}: Props) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const filteredPeople = useMemo(() => {
+    setIsLoading(false);
+
+    return people.filter(person => {
+      return person.name
+        .toLowerCase()
+        .includes(appliedQuery.toLowerCase());
+    });
+  }, [people, appliedQuery]);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, delay),
+    [delay],
```

Comment:
```suggestion
  const applyQuery = useCallback(
    () => debounce(setAppliedQuery, delay),
    [delay],
```
must fix this warning
<img width="648" alt="image" src="https://github.com/mate-academy/react_autocomplete/assets/70220667/73e7e5c9-d20d-445f-af20-5adffea0ef23">